### PR TITLE
LPS-126044 POC: Support all Display Pages in Content Performance Panel

### PR DESCRIPTION
**Pull request by cgoncas**

### CODE REVIEW

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.java [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+import java.util.Optional;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Cristina González
+ */
+@Component(service = AnalyticsReportsInfoItem.class)
+public class LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.java:64]: Should it be LayoutDisplayPageAnalyticsReportsInfoItem?

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/provider/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemObjectProvider.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Cristina González
+ */
+@Component(service = AnalyticsReportsInfoItemObjectProvider.class)
+public class
+	LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemObjectProvider
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/provider/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemObjectProvider.java:32]: Should it be LayoutDisplayPageAnalyticsReportsInfoItemObjectProvider?

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.java [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+		try {
+			LayoutPageTemplateEntry layoutPageTemplateEntry =
+				AssetDisplayPageUtil.getAssetDisplayPageLayoutPageTemplateEntry(
+					layoutDisplayPageObjectProvider.getGroupId(),
+					layoutDisplayPageObjectProvider.getClassNameId(),
+					layoutDisplayPageObjectProvider.getClassPK(),
+					layoutDisplayPageObjectProvider.getClassTypeId());
+
+			return _layoutLocalService.fetchLayout(
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.java:222]: Prevent NPE

**Review by darquesdev — COMMENTED** [review]
wow!! It looks promising @cgoncas, I just left a couple of minor comments. Well done!

**Comment by cgoncas** [comment]
> wow!! It looks promising @cgoncas, I just left a couple of minor comments. Well done!

Hey @darquesdev this is just a POC, so i am not interesting in the review of the names ... What I would want for you is to take a look into the Architecture of this solution.

Have you detected any problem with it? Do you think it is scalable for us?
Thanks!!

**Comment by darquesdev** [comment]
Hi Cris, I didn't understand this way. But the time I spent looking into it, It seemed a very flexible solution  to support all kind of content -adding a new identifier to be able to use `LayoutDisplayPageObjectProvider` as a `AnalyticsReportsObjectProvider`-. I didn't take enough time to have a final conclusion, but I could not find any problem either. I would have liked to discuss a bit more with you, sorry for not having reviewed it before.

**Comment by dgarciasarai** [comment]
Hey @cgoncas, I saw that this Story (LPS-126044) is Ready for QA now. Can we close this POC pull request? Thanks!

**Comment by cgoncas** [comment]
Yeahh @dgarciasarai !!

**Pull request closed without merging**